Source/gui/SNTNotificationManager.m

```diff
@@ -170,10 +190,38 @@ - (void)postDistributedNotification:(SNTMessageWindowController *)pendingMsg {
     kSigningChain : signingChain,
   };
 
-  [dc postNotificationName:@"com.google.santa.notification.blockedeexecution"
-                    object:@"com.google.santa"
-                  userInfo:userInfo
-        deliverImmediately:YES];
+  [self postDistributedNotificationWithUserInfo:userInfo
+                                  notificationName:@"com.google.santa.notification.blockedeexecution"];
+}
+
+- (void)postDistributedNotificationWithEvent:(SNTStoredEvent *)event withBundleHash: (NSString *) bundleHash {
+  NSDictionary *userInfo = @{
+    kFileSHA256: event.fileSHA256 ?: @"",
+    kFilePath : event.filePath ?: @"",
+    kFileBundleName : event.fileBundleName ?: @"",
+    kFileBundleID : event.fileBundleID ?: @"",
+    kFileBundleHash: bundleHash ?: @"",
+  };
+
+  [self postDistributedNotificationWithUserInfo:userInfo
+                                  notificationName:@"com.google.santa.notification.blockedeexecution.bundlehash"];
+}
+
+- (void)postDistributedNotification:(SNTMessageWindowController *)pendingMsg withProgress: (NSUInteger) progressCount {
+  if (![pendingMsg isKindOfClass:[SNTBinaryMessageWindowController class]]) {
+    return;
+  }
+  SNTBinaryMessageWindowController *wc = (SNTBinaryMessageWindowController *)pendingMsg;
+  NSDictionary *userInfo = @{
+    kFileBundleProgress : [NSString stringWithFormat:@"%lu", (unsigned long)progressCount] ?: @"0",
```

Comment:
The `?: @"0"` part here isn't needed, I'm pretty sure it's impossible

Reply:
You are right; removed the extraneous code